### PR TITLE
update accessible locations to be all assigned locations and their de…

### PR DESCRIPTION
**Pull request by mkangia**
…scendants instead of just the primary locations and its descendants

http://manage.dimagi.com/default.asp?242136

@proteusvacuum @esoergel 
We have made these changes for exports but should roll this out earlier since export changes are still under QA and improvements.

cc: @benrudolph

### CODE REVIEW

File: corehq/apps/locations/tests/test_location_queries.py [hunk 1]
```diff
@@ -15,6 +15,9 @@ class BaseTestLocationQuerysetMethods(LocationHierarchyTestCase):
             ('Suffolk', [
                 ('Boston', []),
             ])
+         ]),
```
Comment (dimagimon) [inline corehq/apps/locations/tests/test_location_queries.py]: E121 continuation line under-indented for hanging indent

File: corehq/apps/locations/tests/test_location_queries.py [hunk 2]
```diff
@@ -15,6 +15,9 @@ class BaseTestLocationQuerysetMethods(LocationHierarchyTestCase):
...
+        ('California', [
+             ('Los Angeles', []),
```
Comment (dimagimon) [inline corehq/apps/locations/tests/test_location_queries.py]: E126 continuation line over-indented for hanging indent

File: corehq/apps/locations/tests/test_location_queries.py [hunk 3]
```diff
@@ -79,6 +82,19 @@ def test_location_restricted(self):
             accessible_locs
         )
 
+    def test_location_assigned_and_their_descendants(self):
+        self.web_user.add_to_assigned_locations(self.domain, self.locations['California'])
+        self.restrict_user_to_location(self.web_user)
```
Comment (proteusvacuum) [inline corehq/apps/locations/tests/test_location_queries.py]: nit: I think this name is no longer clear, could be something like `remove_all_locations_access_for_user` or something.

**Review by proteusvacuum — APPROVED** [review]
Kicked build. LGTM on ✅

**Review by proteusvacuum — APPROVED** [review]

File: corehq/apps/locations/tests/test_location_queries.py (hunk 3 above)
Comment (proteusvacuum) [inline corehq/apps/locations/tests/test_location_queries.py]: yeah, that's way better than what I suggested

**Pull request merged**